com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StepInTargetsRequestHandler.java

```diff
@@ -45,37 +55,84 @@ public CompletableFuture<Response> handle(Command command, Arguments arguments,
+        StackFrame stackframe = context.getStackFrameManager().getStackFrame(frameReference);
+        if (stackframe == null) {
+            return Collections.emptyList();
+        }
+
+        Source source = frameReference.getSource() == null ? findSource(stackframe, context) : frameReference.getSource();
+        if (source == null) {
+            return Collections.emptyList();
+        }
+
+        String sourceUri = AdapterUtils.convertPath(source.path, AdapterUtils.isUri(source.path), true);
+        if (sourceUri == null) {
+            return Collections.emptyList();
+        }
+
         ISourceLookUpProvider sourceLookUpProvider = context.getProvider(ISourceLookUpProvider.class);
-        List<MethodInvocation> invocations = sourceLookUpProvider.findMethodInvocations(stackframe);
+        List<MethodInvocation> invocations = sourceLookUpProvider.findMethodInvocations(sourceUri, stackframe.location().lineNumber());
         if (invocations.isEmpty()) {
             return Collections.emptyList();
         }
 
+        long threadId = stackframe.thread().uniqueID();
         List<StepInTarget> targets = new ArrayList<>(invocations.size());
         for (MethodInvocation methodInvocation : invocations) {
             int id = context.getRecyclableIdPool().addObject(threadId, methodInvocation);
             StepInTarget target = new StepInTarget(id, methodInvocation.expression);
-            target.column = methodInvocation.columnStart;
-            target.endColumn = methodInvocation.columnEnd;
-            target.line = methodInvocation.lineStart;
-            target.endLine = methodInvocation.lineEnd;
+            target.column = AdapterUtils.convertColumnNumber(methodInvocation.columnStart,
+                context.isDebuggerColumnsStartAt1(), context.isClientColumnsStartAt1());
+            target.endColumn = AdapterUtils.convertColumnNumber(methodInvocation.columnEnd,
+                context.isDebuggerColumnsStartAt1(), context.isClientColumnsStartAt1());
+            target.line = AdapterUtils.convertLineNumber(methodInvocation.lineStart,
+                context.isDebuggerLinesStartAt1(), context.isClientLinesStartAt1());
+            target.endLine = AdapterUtils.convertLineNumber(methodInvocation.lineEnd,
+                context.isDebuggerLinesStartAt1(), context.isClientLinesStartAt1());
             targets.add(target);
         }
+
+        // TODO remove the executed method calls.
         return targets;
     }
+
+    private Source findSource(StackFrame frame, IDebugAdapterContext context) {
+        ReferenceType declaringType = frame.location().declaringType();
+        String typeName = declaringType.name();
+        String sourceName = null;
+        String sourcePath = null;
+        try {
+            // When the .class file doesn't contain source information in meta data,
+            // invoking Location#sourceName() would throw AbsentInformationException.
```

Comment:
Location or ReferenceType that should be mentioned in the comment ?

Reply:
Nice catch. Yes, we should update the comment to "ReferenceType#sourceName()".